scripts/send-metrics.py

```diff
@@ -189,5 +196,125 @@ def main(args):
         run_target(make_target)
 
 
+def do_opt_in():
+    """
+    Perform the interactive opt-in process.
+    """
+    try:
+        with open(config_path, 'r') as f:
+            config = json.loads(f.read())
+    except FileNotFoundError:
+        config = {}
+
+    # Leave gated off until we're ready for internal invitation (ARCHBOM-1788)
+    if not config.get(FEATURE_KEY):
+        print(
+            "This is not enabled in your environment. "
+            "Reach out to the Arch-BOM team at edX if you want to participate."
+        )
+        return
+
+    if config.get(CONSENT_KEY):
+        print(
+            "It appears you've previously opted-in to metrics reporting. "
+            f"Recorded consent date: {config[CONSENT_KEY]}"
+        )
+        return
+
+    print(
+        "Allow devstack to report anonymized usage metrics?\n"
+        "\n"
+        "This will report usage information to a team at edX (Arch-BOM) so that "
+        "devstack improvements can be planned and evaluated. The exact metrics "
+        "may change over time, but will include information such as the make "
+        "target, a timestamp, duration of command run, the version of devstack, "
+        "and an anonymous user ID. See "
+        "https://openedx.atlassian.net/wiki/spaces/AC/pages/2720432206/Devstack+Metrics "
+        "for more information. You can opt out again at any time.\n"
+        "\n"
+        "Type 'yes' or 'y' to opt in, or anything else to cancel."
+    )
+    answer = input()
+
+    if answer.lower() in ['yes', 'y']:
+        config[CONSENT_KEY] = datetime.now(timezone.utc).isoformat()
+        # Keep their anonymous user ID if they're toggling back and forth
+        config[USER_ID_KEY] = config.get(USER_ID_KEY) or str(uuid.uuid4())
+
+        os.makedirs(config_dir, exist_ok=True)
+        with open(config_path, 'w') as f:
+            f.write(json.dumps(config))
+
+        print(
+            "Thank you for contributing to devstack development! "
+            "You can opt out again at any time with `make metrics-opt-out` "
+            f"or by deleting the file at {config_path}."
+        )
+        # Send record of opt-in so we can tell whether people are
+        # opting in even if they're not running any of the
+        # instrumented commands.
+        send_metrics_to_segment('devstack.opt-in', {}, config)
+    else:
+        print("Not opted-in.")
```

Comment:
Can we avoid the negative?
```suggestion
        print("Opted-out.")
```

Reply:
They aren't actually opted out if they cancel out here -- it doesn't record anything.

Reply:
I see.  Maybe `Cancelled opt-in`?  Or, leave what you have if you prefer.  Either way.

Reply:
I like that.